frontend/src/metabase/admin/upsells/UpsellHosting.tsx

```diff
@@ -0,0 +1,53 @@
+import { t, jt } from "ttag";
+
+const RocketGlobeIllustrationSrc = "app/assets/img/rocket-globe.svg";
+import { useSelector } from "metabase/lib/redux";
+import { getIsHosted } from "metabase/setup/selectors";
+
+import { UpsellCard } from "./components";
+
+export const UpsellHosting = ({ source }: { source: string }) => {
+  const isHosted = useSelector(getIsHosted);
+
+  if (isHosted) {
+    return null;
+  }
+
+  return (
+    <UpsellCard
+      title={t`Minimize maintenance`}
+      campaign="hosting"
+      buttonText={t`Learn more`}
+      buttonLink="https://www.metabase.com/cloud"
+      illustrationSrc={RocketGlobeIllustrationSrc}
+      source={source}
+    >
+      {jt`${(
+        <strong>{t`Migrate to Metabase cloud`}</strong>
+      )} for fast, reliable, and secure deployment.`}
+    </UpsellCard>
+  );
+};
+
+export const UpsellHostingUpdates = ({ source }: { source: string }) => {
+  const isHosted = useSelector(getIsHosted);
+
+  if (isHosted) {
+    return null;
+  }
+
+  return (
+    <UpsellCard
+      title={t`Get automatic updates`}
+      campaign="hosting"
+      buttonText={t`Learn more`}
+      buttonLink="https://www.metabase.com/cloud"
```

Comment:
fwiw, on my local the link went to:

https://www.metabase.com/cloud/?utm_source=product&utm_medium=upsell&utm_campaign=hosting&utm_content=undefined&utm_source_plan=pro-self-hosted

Note the "undefined"

Reply:
ah good catch, the one on the updates page was in JS so it didn't save me from myself!